website/src/theme/Gateway/wallet-selector.js

```diff
@@ -0,0 +1,120 @@
+import { create as createStore } from 'zustand';
+import { distinctUntilChanged, map } from 'rxjs';
+import { providers } from 'near-api-js';
+import { setupWalletSelector } from '@near-wallet-selector/core';
+import { setupModal } from '@near-wallet-selector/modal-ui';
+import { setupMyNearWallet } from '@near-wallet-selector/my-near-wallet';
+import { setupHereWallet } from '@near-wallet-selector/here-wallet';
+
+import { useEffect, useState } from 'react';
+
+export const useWallet = createStore(set => ({
+  signedAccountId: '',
+  logOut: undefined,
+  logIn: undefined,
+  selector: undefined,
+  viewMethod: undefined,
+  callMethod: undefined,
+  setLogActions: ({ logOut, logIn }) => set({ logOut, logIn }),
+  setAuth: ({ signedAccountId }) => set({ signedAccountId }),
+  setMethods: ({ viewMethod, callMethod }) => set({ viewMethod, callMethod }),
+  setStoreSelector: ({ selector }) => set({ selector }),
+}));
+
+export function useInitWallet({ createAccessKeyFor, networkId }) {
+  const setAuth = useWallet(store => store.setAuth);
+  const setLogActions = useWallet(store => store.setLogActions);
+  const setMethods = useWallet(store => store.setMethods);
+  const setStoreSelector = useWallet(store => store.setStoreSelector);
+  const [selector, setSelector] = useState(undefined);
+
+  useEffect(() => {
+    const selector = setupWalletSelector({
+      network: networkId,
+      modules: [setupMyNearWallet(), setupHereWallet()]
```

Comment:
any reason why we are limiting to these two wallets?

Reply:
these are the ones that I tested and know that works